hadoop-ozone/s3gateway/src/test/java/org/apache/hadoop/ozone/s3/metrics/TestS3GatewayMetrics.java

```diff
@@ -110,4 +123,146 @@ public void testHeadObject() throws Exception {
+    oriMetric = metrics.getGetBucketFailure();
+
+    try {
+      // Searching for a bucket that does not exist
+      bucketEndpoint.get("newBucket", null,
+          null, null, 1000, null,
+          null, "random", null,
+          null, null);
+      fail();
+    } catch (OS3Exception e) {
+    }
+
+    curMetric = metrics.getGetBucketFailure();
+    assertEquals(1L, curMetric - oriMetric);
+  }
+
+  @Test
+  public void testCreateBucket() throws Exception {
+
+    long oriMetric = metrics.getCreateBucketSuccess();
+
+    bucketEndpoint.put(bucketName, null,
+        null, null);
+    long curMetric = metrics.getCreateBucketSuccess();
+
+    assertEquals(1L, curMetric - oriMetric);
+
+
+    // Creating an error by trying to create a bucket that already exists
+    oriMetric = metrics.getCreateBucketFailure();
+
+    bucketEndpoint.put(bucketName, null, null, null);
+
+    curMetric = metrics.getCreateBucketFailure();
+    assertEquals(1L, curMetric - oriMetric);
+  }
+
+  @Test
+  public void testDeleteBucket() throws Exception {
+    long oriMetric = metrics.getDeleteBucketSuccess();
+
+    bucketEndpoint.delete(bucketName);
+
+    long curMetric = metrics.getDeleteBucketSuccess();
+    assertEquals(1L, curMetric - oriMetric);
+
+
+    oriMetric = metrics.getDeleteBucketFailure();
+    try {
+      // Deleting a bucket that does not exist will result in delete failure
+      bucketEndpoint.delete(bucketName);
+      fail();
+    } catch (OS3Exception ex) {
+      assertEquals(S3ErrorTable.NO_SUCH_BUCKET.getCode(), ex.getCode());
+      assertEquals(S3ErrorTable.NO_SUCH_BUCKET.getErrorMessage(),
+          ex.getErrorMessage());
+    }
+
+    curMetric = metrics.getDeleteBucketFailure();
+    assertEquals(1L, curMetric - oriMetric);
+  }
+
+  @Test
+  public void testGetAcl() throws Exception {
+    long oriMetric = metrics.getGetAclSuccess();
+
+    Response response =
+        bucketEndpoint.get(bucketName, null, null,
+            null, 0, null, null,
+            null, null, "acl", null);
+    long curMetric = metrics.getGetAclSuccess();
+    assertEquals(HTTP_OK, response.getStatus());
+    assertEquals(1L, curMetric - oriMetric);
+
+
+    oriMetric = metrics.getGetAclFailure();
+    try {
+      // Failing the getACL endpoint by applying ACL on a non-Existent Bucket
+      bucketEndpoint.get("random_bucket", null,
+          null, null, 0, null,
+          null, null, null, "acl", null);
+      fail();
+    } catch (OS3Exception ex) {
+      assertEquals(S3ErrorTable.NO_SUCH_BUCKET.getCode(), ex.getCode());
+      assertEquals(S3ErrorTable.NO_SUCH_BUCKET.getErrorMessage(),
+          ex.getErrorMessage());
+    }
+    curMetric = metrics.getGetAclFailure();
+    assertEquals(1L, curMetric - oriMetric);
+  }
+
+  @Test
+  public void testPutAcl() throws Exception {
+    long oriMetric = metrics.getPutAclSuccess();
+
+    clientStub.getObjectStore().createS3Bucket("b1");
+    InputStream inputBody = TestBucketAcl.class.getClassLoader()
```

Comment:
close this input stream at the end of the test.

Reply:
Thanks will do !!